Reactome management (#11)
gitter-lab/pathway-parameter-advising: README.md

## Dependencies

Pathway parameter advising was written and tested using Python 3.6 and requires the packages `networkx`, `numpy`, and `requests`.

Graphlet decomposition is performed using the PGD library.
The PGD library can be installed from its [github repository](https://github.com/nkahmed/pgd) and complied using `make`.
The script `setupPGD.sh` requires git, and has been provided to aid in installing the PGD library.
However, it is not guaranteed to work and has only been tested on Ubuntu 18.04.

These scripts have only been tested in a Linux environment initially.

## Installation

> To download pathway parameter advising using pip, run the command:

Author: I've updated the readme to mention the general installation strategy.

> > 'pip download --no-deps --no-binary :all: pathwayParameterAdvising`

Reviewer: ```suggestion
> `pip download --no-deps --no-binary :all: pathwayParameterAdvising`
```

> And untar the resulting archive file. 

Reviewer: Should we give the command to help novices?  `tar -xf pathwayParameterAdvising-*.tar.gz` or something similar?

Author: I added this to the README.

Pathway parameter advising can also be downloaded from [the pathway parameter advising github](https://github.com/gitter-lab/pathway-parameter-advising/tree/reactomeManagement). 

> From inside `pathway-parameter-advising` directory, run

Reviewer: When I download from pip, the version is included in the filename.  Mine is `pathwayParameterAdvising-0.1.0`.  If that's expected behavior, we'll need to reword this.  The directory name will depend on the version and will be different if the users downloads from pip or from GitHub.

> From inside `pathway-parameter-advising` directory, run

Author: Changed readme to show this difference between download locations.

> > `python setup.py`

Reviewer: I got an error at this line
```
$ python setup.py
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: no commands supplied
```

Should the command be `python setup.py install` or ` pip install . `?  I believe the pip version is often recommended.

> > `python setup.py`

Author: I changed the readme to instruct users to use `python setup.py install`. `pip install .` gave me an issue where I think it would want me to also download the `.egg` file to install properly (which is an option, but I left it out for now).

to install pathway parameter advising. 

> ## Usage

Reviewer: The better option may be to note these are the Python arguments but refer users to the `bin` directory's `runPPA.sh` script as the most common way to use PPA.

> ## Usage

Author: I updated the usage instructions to make it clear how to use the bash script and still mention ppa.py.

>
>  --nameMap NAMEMAP     Either a file mapping generated pathway fileNames to parameter values, "stripped" to exclude the directory and extension from the filename, or "fileName" to use raw file names. Optional, default = stripped.
>
>  --saveGraphlets       If set, will save graphlet distributions as pickled dictionaries.
>
>  --verbose             If set, will print intermediate status updates. Optional, default = False.

## Examples

`bin/runPPA.sh` runs pathway parameter advising on any set of sif or edgelist networks.

> It will attempt to install the pgd library and setup the reactome pathway graphlets automatically. 

Reviewer: `runPPA.sh` no longer installs PGD

Author: Removed this where runPPA.sh is mentioned.

> The following commands will run `runPPA.sh` with the included `Wnt` and `Prolactin` datasets from the `bin` directory, where `lib/pgd` is installation location of the PGD library:

Reviewer: ```suggestion
The following commands will run `runPPA.sh` with the included `Wnt` and `Prolactin` datasets from the `bin` directory, where `../lib/pgd` is installation location of the PGD library:
```

> `bash runPPA.sh ../data/Wnt wnt_ranking.txt ../lib/pgd`
>
> `bash runPPA.sh ../data/Prolactin prolactin_ranking.txt ../lib/pgd`

`bin/runNetBoxIL2.sh` runs pathway parameter advising for the precomputed graphlet files for NetBox IL2 pathways using Reactome reference pathways.

> These scripts must be run from the `bin` directory.

Reviewer: ```suggestion
These scripts must be run from the `bin` directory, and file path arguments are relative to the `bin` directory.
```

## Graphlet Creation
Graphlet decomposition files are created with  the [Parallel Graphlet Decomposition library](http://nesreenahmed.com/graphlets/).
Files are the piped output from the pgd script: `./pgd -f inputGraphFile >> graphletOutputFile.gOut`.

## Other Scripts

> `bin/setupPGD.sh` installs the PGD library into the `lib` directory, which is created if none exists.

Reviewer: ```suggestion
`bin/setupPGD.sh` installs the PGD library into the `lib` directory, which is created if it does not exist.
```

PGD is cloned from its [github repository](https://github.com/nkahmed/pgd) and complied using `make`.
It can then be run from the base pathway-parameter-advising directory as `lib/pgd/pgd -f inputGraphFile`.

> `bin/updateReactome.sh` downloads the latest version of all human reactome pathways from [Pathway Commons](https://www.pathwaycommons.org/) and performs graphlet decomposition on them. Requires the location of the PGD library as a command line argument. 

Reviewer: ```suggestion
`bin/updateReactome.sh` downloads the latest version of all human Rreactome pathways from [Pathway Commons](https://www.pathwaycommons.org/) and performs graphlet decomposition on them.
Requires the location of the PGD library as a command line argument. 
```
